pandas/core/indexes/interval.py

```diff
@@ -97,6 +98,42 @@ def _new_IntervalIndex(cls, d):
     return cls.from_arrays(**d)
 
 
+class setop_check:
```

Comment:
Can you use PEP8 naming conventions for a classes, i.e. `SetopCheck` and add a parameters section to the docstring for `op_name`?